### PR TITLE
adds calculation of global values to ljfunctor's soa functor

### PR DESCRIPTION
**Pull request by SteffenSeckler**
# Description

adds calculation of global values to ljfunctor's soa functor

## Related Pull Requests

- #102 
- #113 

## Resolved Issues # (issue)

- [x] #100 (SoA part, not yet verlet)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Unit tests
- [x] Using ls1

### CODE REVIEW

File: examples/md-flexible/strongscaling.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-
+export LC_NUMERIC=en_US.UTF-8
```
Comment (FG-TUM) [inline examples/md-flexible/strongscaling.sh]: We will also need this in the `measurePerf.sh` script

File: src/autopas/pairwiseFunctors/LJFunctor.h [hunk 2]
```diff
@@ -132,18 +133,33 @@ class LJFunctor : public Functor<Particle, ParticleCell, typename Particle::SoAA
     double *const __restrict__ fxptr = soa.template begin<Particle::AttributeNames::forceX>();
     double *const __restrict__ fyptr = soa.template begin<Particle::AttributeNames::forceY>();
     double *const __restrict__ fzptr = soa.template begin<Particle::AttributeNames::forceZ>();
+    const double cutoffsquare = _cutoffsquare, epsilon24 = _epsilon24, sigmasquare = _sigmasquare, shift6 = _shift6;
```
Comment (FG-TUM) [inline src/autopas/pairwiseFunctors/LJFunctor.h]: document why you did this

File: src/autopas/pairwiseFunctors/LJFunctor.h [hunk 3]
```diff
@@ -203,14 +242,36 @@ class LJFunctor : public Functor<Particle, ParticleCell, typename Particle::SoAA
...
+      // Checks if the cell is a halo cell, if it is, we skip it.
+      // We cannot do this in normal cases (where we do not calculate globals), as _lowCorner and _highCorner are not
+      // set. (as of 23.11.2018)
+      isHaloCell1 |= x1ptr[0] < _lowCorner[0] || x1ptr[0] >= _highCorner[0];
+      isHaloCell1 |= y1ptr[0] < _lowCorner[1] || y1ptr[0] >= _highCorner[1];
+      isHaloCell1 |= z1ptr[0] < _lowCorner[2] || z1ptr[0] >= _highCorner[2];
+      isHaloCell2 |= x2ptr[0] < _lowCorner[0] || x2ptr[0] >= _highCorner[0];
+      isHaloCell2 |= y2ptr[0] < _lowCorner[1] || y2ptr[0] >= _highCorner[1];
+      isHaloCell2 |= z2ptr[0] < _lowCorner[2] || z2ptr[0] >= _highCorner[2];
+      /*if(_duplicatedCalculations and isHaloCell1 and isHaloCell2){
```
Comment (FG-TUM) [inline src/autopas/pairwiseFunctors/LJFunctor.h]: document why you commented this out

File: src/autopas/pairwiseFunctors/LJFunctor.h [hunk 4]
```diff
@@ -203,14 +242,36 @@ class LJFunctor : public Functor<Particle, ParticleCell, typename Particle::SoAA
     double *const __restrict__ fy2ptr = soa2.template begin<Particle::AttributeNames::forceY>();
     double *const __restrict__ fz2ptr = soa2.template begin<Particle::AttributeNames::forceZ>();
 
+    bool isHaloCell1 = false;
+    bool isHaloCell2 = false;
+    if (calculateGlobals) {
+      // Checks if the cell is a halo cell, if it is, we skip it.
+      // We cannot do this in normal cases (where we do not calculate globals), as _lowCorner and _highCorner are not
```
Comment (FG-TUM) [inline src/autopas/pairwiseFunctors/LJFunctor.h]: move this comment to the if.

File: tests/testAutopas/LJFunctorTest.cpp [hunk 5]
```diff
@@ -58,6 +58,108 @@ TEST_F(LJFunctorTest, testAoSFunctorNoGlobalsN3) {
   testAoSNoGlobals(newton3);
 }
 
+void LJFunctorTest::testSoANoGlobals(bool newton3, CellInteractionType interactionType) {
```
Comment (FG-TUM) [inline tests/testAutopas/LJFunctorTest.cpp]: Zitat: "Könnte man kommentieren" <- !!!

File: tests/testAutopas/LJFunctorTest.cpp [hunk 6]
```diff
@@ -58,6 +58,108 @@ TEST_F(LJFunctorTest, testAoSFunctorNoGlobalsN3) {
...
+      functor.SoAFunctor(cell1._particleSoABuffer, newton3);
+      break;
+    case CellInteractionType::pair:
+      functor.SoAFunctor(cell1._particleSoABuffer, cell2._particleSoABuffer, newton3);
+      break;
+  }
+  functor.SoAExtractor(cell1, cell1._particleSoABuffer);
+  functor.SoAExtractor(cell2, cell2._particleSoABuffer);
+
+  auto f1one = cell1.begin()->getF();
```
Comment (FG-TUM) [inline tests/testAutopas/LJFunctorTest.cpp]: props for the name

File: examples/md-flexible/strongscaling.sh (hunk 1 above)
Comment (SteffenSeckler) [reply examples/md-flexible/strongscaling.sh]: you will do this!

**Pull request merged**